### PR TITLE
Consolidate list remote users api

### PR DESCRIPTION
**Pull request by AlexVelezLl**
## Summary

Consolidates the logic of listing remote users from the user_profile and setup_wizard plugins, and migrates them to core auth so that they are accessible by multiple plugins.

There were some differences between both endpoints, and the following decisions were made to merge them:
* The setup wizard endpoint returned an object { admin, students }, while the user_profile endpoint returned a list of users. It was decided to keep the response the same as that of user_profile as it was more flexible, and in the setup wizard to make the admin, students filter on the frontend instead of the backend.
* The setup wizard endpoint returns a 403 error if there are authentication errors, but the user_profile endpoint returns a 200 code, but with an error object. The 403 error was maintained, and where it is called to the user_profile endpoint in the frontend the error handling was updated.
* The setup_wizard endpoint gets a "facility_id" param, while the user_profile endpoint gets a "facility" param. The naming of "facility_id" was preferred.
* The user_profile endpoint validated the input data, and the setup_wizard endpoint did not. It was preferred to validate it as the user_profile.

## Reviewer guidance
* Check the flow of importing users as administrator of the setup wizard.
* Check the flow of changing facilities for an "on my own" user, and merging users using administrator credentials.

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/discovery/api.py [hunk 1]
```diff
@@ -77,6 +77,7 @@ class StaticNetworkLocationViewSet(NetworkLocationViewSet):
 
 class NetworkLocationFacilitiesView(BaseValuesViewset):
     queryset = NetworkLocation.objects.all()
+    values = ()
```
Comment (AlexVelezLl) [inline kolibri/core/discovery/api.py]: This is just a quick fix I found to be able to list the devices. Should be removed before merge.

**Review by rtibbles — COMMENTED** [review]
Changes make sense, and being able to reuse the same API endpoint with just some minor tweaks of how they are handled in Javascript is a neat way to do this with minimal changes.

Having more tests, and also using the API endpoints more consistently would be preferable in the future, but this does well to do the minimal required change.

**Comment by pcenov** [comment]
Hi @AlexVelezLl - I confirm that there are no issues when importing learners as an administrator in the setup wizard or when changing facilities for an "On my own" learner. However when I attempt to merge a learner that was created on the actual "On my own" device then when I try to migrate the learner either by using his password or by using an admin account I am getting permission denied 403 errors in the console and the learner cannot be migrated:

https://github.com/user-attachments/assets/ca56ec9f-968c-417b-809c-1cc73f6fa8c5

[LogsDB.zip](https://github.com/user-attachments/files/16549996/LogsDB.zip)

**Comment by AlexVelezLl** [comment]
Hi @pcenov! I have solved the mentioned issues both when mergin the self user and with an admin:

Screenshot from 2024-08-08 15-42-38

@rtibbles I have updated the RemoteFacilityUserAuthenticatedViewset so if the queried user is not an admin we just return the user: https://github.com/learningequality/kolibri/pull/12321/commits/98f2f550c7f2f931fab16f636dde88b810a8d217.

Also, since [this commit](https://github.com/learningequality/kolibri/commit/c8e357530cace57aa2c61151910b10f36147343e) the [IsSelf permission](https://github.com/learningequality/kolibri/blob/5582278a1d4179bc2191e9545275a4d93b647bf1/kolibri/plugins/user_profile/tasks.py#L128) was not working ok because `local_user_id` was not in `kwargs`. I tried to solve this [here](https://github.com/learningequality/kolibri/pull/12321/commits/5e36156e92ed8eadb62d2f187dd51616f75802e5), would love to hear if you have any comment about this :open_hands:.

**Comment by rtibbles** [comment]
I think rather than mess with the signature of the task function, it might be simpler just to update how the IsSelf references the args/kwargs?

```
class IsSelf(BasePermission):
    def user_can_run_job(self, user, job):
        return user.id == job.args[1] or user.id == job.kwargs.get("remote_user_pk", None)

    def user_can_read_job(self, user, job):
        return user.id == job.args[1] or user.id == job.kwargs.get("remote_user_pk", None)
```

**Comment by pcenov** [comment]
Thanks @AlexVelezLl - I confirm that everything is working correctly now!

**Comment by AlexVelezLl** [comment]
Thank you @pcenov!  @rtibbles would you give it one last look, please? :open_hands:.

**Review by rtibbles — APPROVED** [review]

**Pull request merged**